web/src/app/core/services/sorting.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+
+/**
+ * Sorting service
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class SortingService {
+  /**
+   * This function is used for sorting the list by date field
+   */
+  sortList(list: any, field: any): void {
```

Comment:
Can `list` be an `array` and `field` be a `string`?

Reply:
List can't be an array , as its not fixed which type of model it can be. It can be release model or issue model but yes field can be of type string here.

Reply:
Can a generic be used, something like...

```typescript
sortList<T>(list: Array<T>, field: string): void { }
```

Reply:
Its giving this error . 
![image](https://user-images.githubusercontent.com/10898595/59786524-13458500-92e5-11e9-8b54-27ed927e8bc7.png)

When I am using the suggestion given by quick fix to use T[]. Its giving another error . Not able to find T

Reply:
You need it before the method signature also `<T>(`

```typescript
sortListByDate<T>(list: Array<T>, field: string): void { }
```

Reply:
`sortListByDate<T>(list: T[], field: string):`

Reply:
Looks good 👍

Reply:
It worked for me. Will be fixed as part of next PR 
https://github.com/DashboardHub/PipelineDashboard/pull/1327